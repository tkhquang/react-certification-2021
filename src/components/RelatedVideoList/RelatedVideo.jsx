import React from 'react';
import { Link } from 'react-router-dom';

import {
  StyledRelatedVideo,
  StyledRelatedVideoCover,
  StyledRelatedVideoTitle,
} from './RelatedVideo.styled';

const RelatedVideo = ({ title, thumbnails, videoId }) => {
  return (
    <Link as={StyledRelatedVideo} to={`/${videoId}`}>
      <StyledRelatedVideoCover src={thumbnails.medium.url} alt={title} />
      <StyledRelatedVideoTitle>{title}</StyledRelatedVideoTitle>
    </Link>
  );
};

export default RelatedVideo;
