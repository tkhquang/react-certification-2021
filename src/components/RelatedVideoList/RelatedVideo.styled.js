import styled from 'styled-components';

export const StyledRelatedVideo = styled.div`
  display: flex;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
  overflow: hidden;
  cursor: pointer;
`;

export const StyledRelatedVideoCover = styled.img`
  color: grey;
  height: 90px;
  width: 120px;
  min-height: 90px;
  min-width: 120px;
  object-fit: cover;
  background-color: grey;
`;

export const StyledRelatedVideoTitle = styled.div`
  font-size: 1rem;
  line-height: 1.25rem;
  font-weight: semi-bold;
  margin: 0.75rem;
  height: calc(1.25rem * 2);

  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
`;
