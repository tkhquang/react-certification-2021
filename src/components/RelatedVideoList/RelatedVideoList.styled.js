import styled, { css } from 'styled-components';

export const StyledRelatedVideoList = styled.div`
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  grid-auto-rows: 120px;

  ${({ theme }) => {
    const { spaces } = theme;
    return css`
      gap: ${spaces.md};
    `;
  }}
`;
