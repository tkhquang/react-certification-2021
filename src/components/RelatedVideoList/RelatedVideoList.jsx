import React from 'react';

import RelatedVideo from './RelatedVideo';
import { StyledRelatedVideoList } from './RelatedVideoList.styled';

const RelatedVideoList = ({ videos }) => {
  return (
    <StyledRelatedVideoList>
      {videos.map(({ etag, id, snippet }) => {
        return <RelatedVideo key={etag} {...snippet} videoId={id.videoId} />;
      })}
    </StyledRelatedVideoList>
  );
};

export default RelatedVideoList;
