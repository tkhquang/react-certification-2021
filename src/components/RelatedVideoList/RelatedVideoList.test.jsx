import React from 'react';
import renderer from 'react-test-renderer';

import { TestAppContainer, mocks } from '../../test';

import RelatedVideoList from './RelatedVideoList';

describe('RelatedVideoList', () => {
  test('renders video list correctly', () => {
    const tree = renderer
      .create(
        <TestAppContainer>
          <RelatedVideoList videos={mocks.dummyData.items} />
        </TestAppContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
