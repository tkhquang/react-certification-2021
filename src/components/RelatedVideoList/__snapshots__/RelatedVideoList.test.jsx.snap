// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedVideoList renders video list correctly 1`] = `
<div
  className="sc-hKFxyN jrcDfP"
>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/undefined"
    onClick={[Function]}
  >
    <img
      alt="Wizeline"
      className="sc-gtsrHT dzxLSG"
      src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/nmXMgqjQzls"
    onClick={[Function]}
  >
    <img
      alt="Video Tour | Welcome to Wizeline Guadalajara"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Video Tour | Welcome to Wizeline Guadalajara
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/HYyRZiwBWc8"
    onClick={[Function]}
  >
    <img
      alt="Wizeline Guadalajara | Bringing Silicon Valley to Mexico"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/HYyRZiwBWc8/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline Guadalajara | Bringing Silicon Valley to Mexico
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/Po3VwR_NNGk"
    onClick={[Function]}
  >
    <img
      alt="Wizeline hace sentir a empleados como en casa"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/Po3VwR_NNGk/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline hace sentir a empleados como en casa
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/7PtYNO6g7eI"
    onClick={[Function]}
  >
    <img
      alt="We Are Wizeline"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/7PtYNO6g7eI/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      We Are Wizeline
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/YuW0CE_8i1I"
    onClick={[Function]}
  >
    <img
      alt="Wizeline Thrives in Mexico City"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/YuW0CE_8i1I/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline Thrives in Mexico City
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/CHzlSGRvWPs"
    onClick={[Function]}
  >
    <img
      alt="Wizeline"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/CHzlSGRvWPs/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/cjO2fJy8asM"
    onClick={[Function]}
  >
    <img
      alt="A Day in the Life of an Engineering Manager at Wizeline"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/cjO2fJy8asM/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      A Day in the Life of an Engineering Manager at Wizeline
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/zClI9OjgKXM"
    onClick={[Function]}
  >
    <img
      alt="Wizeline Technical Writing Academy | Featuring Eduardo Ocejo"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/zClI9OjgKXM/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/8bz9R61oY5o"
    onClick={[Function]}
  >
    <img
      alt="Silicon Valley en México"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/8bz9R61oY5o/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Silicon Valley en México
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/7dJFraOqcoQ"
    onClick={[Function]}
  >
    <img
      alt="Why Wizeline? featuring Juan Pablo Villa in Mexico City"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/7dJFraOqcoQ/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Why Wizeline? featuring Juan Pablo Villa in Mexico City
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/w-Qwc_XJrWc"
    onClick={[Function]}
  >
    <img
      alt="Wizeline&#39;s 2020 Year in Review"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/w-Qwc_XJrWc/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline&#39;s 2020 Year in Review
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/rjir_cHTl5w"
    onClick={[Function]}
  >
    <img
      alt="Why Wizeline? featuring Hugo Lopez in Mexico City"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/rjir_cHTl5w/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Why Wizeline? featuring Hugo Lopez in Mexico City
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/DcFK1x3NHGY"
    onClick={[Function]}
  >
    <img
      alt="Why Wizeline? (We&#39;re Hiring in Mexico!)"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/DcFK1x3NHGY/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Why Wizeline? (We&#39;re Hiring in Mexico!)
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/3BzYWAqZgFw"
    onClick={[Function]}
  >
    <img
      alt="Why Wizeline? featuring Oswaldo Herrera in Mexico City"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/3BzYWAqZgFw/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Why Wizeline? featuring Oswaldo Herrera in Mexico City
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/3KVFmT-Tp2w"
    onClick={[Function]}
  >
    <img
      alt="Caso de Éxito AWS: Wizeline [Spanish]"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/3KVFmT-Tp2w/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Caso de Éxito AWS: Wizeline [Spanish]
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/aKuPmY2m1Ro"
    onClick={[Function]}
  >
    <img
      alt="Wizeline&#39;s 2019 Year in Review"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/aKuPmY2m1Ro/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline&#39;s 2019 Year in Review
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/24sTHUyWhRM"
    onClick={[Function]}
  >
    <img
      alt="[1 of 2] Wizeline CEO shares career lessons from Google"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/24sTHUyWhRM/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      [1 of 2] Wizeline CEO shares career lessons from Google
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/IxGc1gSqB3A"
    onClick={[Function]}
  >
    <img
      alt="Wizeline Data Engineering featuring Tania Reyes"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/IxGc1gSqB3A/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline Data Engineering featuring Tania Reyes
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/NP1gAnbeNno"
    onClick={[Function]}
  >
    <img
      alt="Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/NP1gAnbeNno/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Wizeline Querétaro | Mexico&#39;s New Knowledge Economy (We&#39;re hiring!)
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/F6Krwu6lUc8"
    onClick={[Function]}
  >
    <img
      alt="Miriam Godinez | Women in Leadership at Wizeline"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/F6Krwu6lUc8/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Miriam Godinez | Women in Leadership at Wizeline
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/RFq7gfvhtCk"
    onClick={[Function]}
  >
    <img
      alt="Welcome Back to Wizeline Vietnam | Extended Version"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/RFq7gfvhtCk/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Welcome Back to Wizeline Vietnam | Extended Version
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/E1Vq_A3WKK8"
    onClick={[Function]}
  >
    <img
      alt="How does Wizeline work?"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/E1Vq_A3WKK8/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      How does Wizeline work?
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/ZmkslANDz0Q"
    onClick={[Function]}
  >
    <img
      alt="12 Wishes from Wizeline | Happy Holidays 2019"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/ZmkslANDz0Q/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      12 Wishes from Wizeline | Happy Holidays 2019
    </div>
  </a>
  <a
    className="sc-bdnxRM dBdCUa"
    href="/Nss3EmTDD3s"
    onClick={[Function]}
  >
    <img
      alt="Why Wizeline?"
      className="sc-gtsrHT dzxLSG"
      src="https://i.ytimg.com/vi/Nss3EmTDD3s/mqdefault.jpg"
    />
    <div
      className="sc-dlnjwi kGsIqP"
    >
      Why Wizeline?
    </div>
  </a>
</div>
`;
